src/poetry/core/packages/file_dependency.py

```diff
@@ -43,10 +35,40 @@ def __init__(
             optional=optional,
             allows_prereleases=True,
             source_type="file",
-            source_url=self._full_path.as_posix(),
+            source_url=full_path.as_posix(),
             extras=extras,
         )
 
+    def _get_full_path(self, raise_if_invalid: bool, name: str) -> Path:
+        if self._full_path is not None:
+            return self._full_path
```

Comment:
This little bit of caching means that if we already checked in `__init__` we don't need to check again. We could also cache the negative case but it doesn't seem worth the complexity IMO.